tests/src/main/java/net/neoforged/neoforge/debug/attachment/AttachmentTests.java

```diff
@@ -158,4 +167,73 @@ static void dynamicDataContentSerialization(DynamicTest test, RegistrationHelper
             helper.succeed();
         });
     }
+
+    @TestHolder(description = "Tests serialization for an empty object", enabledByDefault = true)
+    static void pseudoEmptyAttachmentSerialization(DynamicTest test, RegistrationHelper reg) {
+        class EmptyObject {
+            @Nullable
+            public Integer value;
+
+            public EmptyObject(Optional<Integer> value) {
+                this.value = value.orElse(null);
+            }
+
+            @Override
+            public boolean equals(Object obj) {
+                if (!(obj instanceof EmptyObject empty)) return false;
+                return Objects.equals(this.value, empty.value);
+            }
+        }
+
+        class HolderTest extends AttachmentHolder implements ValueIOSerializable {
+            @Override
+            public void serialize(ValueOutput output) {
+                this.serializeAttachments(output);
+            }
+
+            @Override
+            public void deserialize(ValueInput input) {
+                this.deserializeAttachments(input);
+            }
+        }
+
+        var nullableAttachment = reg.attachments()
+                .register("nullable_attachment", () -> AttachmentType.builder(() -> new EmptyObject(Optional.of(1)))
+                        .serialize(RecordCodecBuilder.mapCodec(
+                                instance -> instance.group(
+                                        Codec.INT.optionalFieldOf("value").forGetter(obj -> Optional.ofNullable(obj.value))).apply(instance, EmptyObject::new)))
```

Comment:
I wonder if an optional that would default to 0 would spare you from all the Optional stuff, both here and in the class above.

Reply:
Honestly, probably.